cloudferrylib/base/resource.py

```diff
@@ -36,8 +41,18 @@ def save(self):
     def restore(self):
         pass
 
-    def wait_for_status(self, id_obj, status, limit_retry=60):
-        pass
+    def wait_for_status(self, res_id, get_status, wait_status, limit_retry=60, warn_only=False):
```

Comment:
Even though logic follows earlier versions of the code, I think it is not fully correct. Wait for status should wait no longer than configured value (say, 5 minutes), and raise error or print log, it should not rely on number of retries. Something like
```
def wait_for_status(openstack_object, expected_status)
    delay = 1
    while delay < config.timeout:
        if openstack_object_status != expected_status:
            break
        sleep(delay)
        delay *= 2
    else:
        raise Timeout()
```

Also, instead of handling raise or not raise an error as part of a single method, let's better have two different methods which work differently:
```
def wait_for_status()
"""Raises error"""
    <implementation>

def try_wait_for_status()
"""Not raises error"""
    try:
        wait_for_status()
    except Timeout as e:
        LOG.warning(e)
```